api/src/queries/project/project-update-queries.ts

```diff
@@ -391,3 +391,35 @@ export const putProjectFundingSourceSQL = (
 
   return sqlStatement;
 };
+
+/**
+ * SQL query to update the publish status of a project.
+ *
+ * @param {number} projectId
+ * @param {boolean} publish
+ * @returns {SQLStatement} sql query object
+ */
+export const updateProjectPublishStatusSQL = (projectId: number, publish: boolean): SQLStatement | null => {
+  defaultLog.debug({ label: 'updateProjectPublishStatusSQL', message: 'params', projectId, publish });
+
+  if (!projectId) {
+    return null;
+  }
+
+  const sqlStatement: SQLStatement = SQL`
+      UPDATE project
+      SET publish_timestamp = ${publish ? 'now()' : null}
+      WHERE id = ${projectId}
```

Comment:
If the project is already published, we don't want to update the row with a new publish_timestamp.
Can we add an extra condition here to check for "and publish_timetamp = null" (or w/e the correct syntax is).

Reply:
Not sure what the return value is in this case though, might impact the response checks in the endpoint code?